ion-hash/tests/ion_hash_tests.rs

```diff
@@ -97,30 +128,22 @@ fn test_case<E: Element>(ion: &E, strukt: &E) -> IonResult<()> {
 
         match method {
             "update" => {
-                let first = recorded_updates
-                    .nth(0)
-                    .expect("no more calls to `update` were made");
-
-                assert_eq!(&bytes[..], &[first]);
+                // TODO: We currently don't assert on intermediate updates. It's
+                // not clear if this is actually valuable, other than helping
+                // diagnose bugs.
             }
             "digest" => {
-                assert_eq!(bytes, recorded_digest);
+                for (i, byte) in bytes.iter().enumerate() {
+                    assert_eq!(
+                        *byte, result[i],
+                        "case: {}; byte {} failed to match",
+                        test_case_name, i
+                    );
+                }
```

Comment:
Are you comparing each byte, byte for byte to make it easier to determine about where the problem is?

Reply:
Hold my beer..

So. The ion-hash-tests aren't actually doing any digesting (because that's not the point). Usually, a digest implementation has a fixed size, so the digest trait lets you pick the output size. But, we don't have a fixed size because our final digest is just the u8s we fed to the digester. Which means that test-1 and test-2 have different lens. To play with the generic array trait, I picked a N that is bigger than the largest test case. But.. that means that we have trailing 0s in our array. Or, put another way, `result` is exactly equal to `bytes` plus 0s to the end of the array.

I'll add a comment.